src/frontend/src/components/Posts/Posts.jsx

```diff
@@ -12,11 +14,65 @@ const useStyles = makeStyles({
   },
 });
 
-const Posts = ({ posts }) => {
+export default function Posts({ posts }) {
+  const [initNumPosts, setInitNumPosts] = useState(0);
+  const [currentNumPosts, setCurrentNumPosts] = useState(0);
+  const savedCallback = useRef();
+  const { telescopeUrl } = useSiteMetaData();
   const classes = useStyles();
 
+  async function getPostsCount() {
+    try {
+      const res = await fetch(`${telescopeUrl}/posts`, { method: 'HEAD' });
+      if (!res.ok) {
+        throw new Error(res.statusText);
+      }
+      return res.headers.get('x-total-count');
+    } catch (error) {
+      console.log(error);
+    }
+    return null;
+  }
+
+  function callback() {
+    getPostsCount()
+      .then(setCurrentNumPosts)
+      .catch((error) => console.log(error));
+  }
+
+  useEffect(() => {
+    savedCallback.current = callback;
+  });
+
+  // Get the current + initial posts count when page loads
+  useEffect(() => {
+    async function setPostsInfo() {
+      try {
+        await Promise.all([
+          setInitNumPosts(await getPostsCount()),
+          setCurrentNumPosts(await getPostsCount()),
+        ]);
+      } catch (error) {
+        console.log({ error });
+      }
+    }
+    setPostsInfo();
+  }, []);
+
+  useEffect(() => {
+    function getCurrentNumPosts() {
+      savedCallback.current();
+    }
+
+    savedCallback.current = callback;
+    // Polls every 5 minutes
+    const interval = setInterval(getCurrentNumPosts, 5 * 60 * 1000);
+    return () => clearInterval(interval);
+  }, [currentNumPosts]);
+
   return (
     <Container className={classes.root}>
+      {currentNumPosts !== initNumPosts ? <CustomizedSnackBar posts={currentNumPosts} /> : null}
```

Comment:
I think the only change that was needed was supposed to be somewhere here with the following line
{posts.length > 0 ? (do mapping here) : null}`